src/screens/Asset/ReceiveModal.js

```diff
@@ -117,34 +119,56 @@ const ReceiveModal = ({
       <ContentWrapper forceInset={{ top: 'never', bottom: 'always' }}>
         <InfoView>
           {!!ensName && (
-            <TextWithCopy textToCopy={ensName} toastText={t('toast.ensNameCopiedToClipboard')} iconColor={colors.link}>
+            <TextWithCopy
+              textToCopy={ensName}
+              toastText={t('toast.ensNameCopiedToClipboard')}
+              iconColor={colors.link}
+              textStyle={styles.ensName}
+              adjustsFontSizeToFit
+              numberOfLines={1}
+            >
               {ensName}
             </TextWithCopy>
           )}
           {ensName ? (
-            <WalletAddress>{address}</WalletAddress>
+            <WalletAddress numberOfLines={1} adjustsFontSizeToFit>
+              {address}
+            </WalletAddress>
           ) : (
             <TextWithCopy
               toastText={t('toast.addressCopiedToClipboard')}
               textToCopy={address}
-              textStyle={{ color: colors.basic030 }}
+              textStyle={[styles.address, { color: colors.secondaryText }]}
               iconColor={colors.link}
+              adjustsFontSizeToFit
+              numberOfLines={1}
             >
               {address}
             </TextWithCopy>
           )}
         </InfoView>
-        {!!address && <QRCodeWrapper><QRCodeWithTheme value={address} size={104} /></QRCodeWrapper>}
-        <WarningText center small>
-          {t('paragraph.cautionMessage', {
-            chain: t('chains.ethereum'),
-            mediumText: true,
-            color: colors.recieveModalWarningText,
-          })}{' '}
-          <Text color={colors.link} onPress={() => navigation.navigate(ETHERSPOT_DEPLOYMENT_INTERJECTION)}>
-            {t('paragraph.withCaution')}
-          </Text>
-        </WarningText>
+        {!!address && (
+          <QRCodeWrapper>
+            <QRCodeWithTheme value={address} size={104} />
+          </QRCodeWrapper>
+        )}
+        {activeAccount?.type !== ACCOUNT_TYPES.ARCHANOVA_SMART_WALLET && (
+          <WarningText style={{ marginTop: spacing.medium }} center small>
```

Comment:
1. modern `Text` does not accept `center` nor `small` props. Pls remove them as they have no effect.

```suggestion
          <WarningText style={{ marginTop: spacing.medium }}>
```

2. Pls extract `{ marginTop: spacing.medium }` as `styles.signleAddressInfo` style.

Reply:
Done